source/includes/fundamentals/code-snippets/builders/AggBuilders.java

```diff
@@ -0,0 +1,286 @@
+        unwind("$sizes", new UnwindOptions().includeArrayIndex("position"));
+        // end unwindIndex
+    }
+
+    private void unwindPreserveStage() {
+        // begin unwindPreserve
+        unwind("$sizes", new UnwindOptions().preserveNullAndEmptyArrays(true));
+        // end unwindPreserve
+    }
+
+    private void unwindStage() {
+        // begin unwindStage
+        unwind("$sizes");
+        // end unwindStage
+    }
+
+    private void groupStage() {
+        // begin group
+        group("$customerId", Accumulators.sum("totalQuantity", "$quantity"));
+        // end group
+    }
+
+    private void advancedLookupStage() {
+        database = mongoClient.getDatabase("agg-builders");
+        collection = database.getCollection("warehouses");
+        collection.drop();
+        collection.insertMany(asList(
+                new Document("_id", 1).append("stock_item", "almonds").append("warehouse", "A").append("instock", 120),
+                new Document("_id", 2).append("stock_item", "pecans").append("warehouse", "A").append("instock", 80),
+                new Document("_id", 3).append("stock_item", "almonds").append("warehouse", "B").append("instock", 60),
+                new Document("_id", 4).append("stock_item", "cookies").append("warehouse", "B").append("instock", 40),
+                new Document("_id", 5).append("stock_item", "cookies").append("warehouse", "A").append("instock", 80)));
+        collection = database.getCollection("orders");
+        collection.drop();
+        collection.insertMany(
+                asList(new Document("_id", 1).append("item", "almonds").append("price", 12).append("ordered", 2),
+                        new Document("_id", 2).append("item", "pecans").append("price", 20).append("ordered", 1),
+                        new Document("_id", 3).append("item", "cookies").append("price", 10).append("ordered", 60)));
+        // begin advanced lookup
+        List<Variable<String>> variables = asList(new Variable<>("order_item", "$item"),
+                new Variable<>("order_qty", "$ordered"));
+
+        List<Bson> pipeline = asList(
+                match(Filters.expr(new Document("$and",
+                        asList(new Document("$eq", asList("$$order_item", "$stock_item")),
+                                new Document("$gte", asList("$instock", "$$order_qty")))))),
+                project(Projections.fields(Projections.exclude("stock_item"), Projections.excludeId())));
+
+        MongoCursor<Document> cursor = collection
+                .aggregate(asList(lookup("warehouses", variables, pipeline, "stockdata"))).cursor();
+        // end advanced lookup
+        cursor.forEachRemaining(doc -> System.out.println(doc.toJson()));
+        database = mongoClient.getDatabase("sample_mflix");
+        collection = database.getCollection("movies");
+    }
+
+    private void basicLookupStage() {
+        // begin basic lookup
+        lookup("comments", "_id", "movie_id", "joinedcomments");
```

Comment:
```suggestion
        lookup("comments", "_id", "movie_id", "joined_comments");
```
Your bullet has an underscore.